### PR TITLE
[Executor] Support image type in executor

### PR DESCRIPTION
**Pull request by lumoslnt**
# Description

Support image type in executor.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by thy09 — APPROVED** [review]

**Review by zzn2 — DISMISSED** [review]

**Review by thy09 — APPROVED** [review]

**Review by thy09 — DISMISSED** [review]

**Review by thy09 — DISMISSED** [review]

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 1]
```diff
@@ -293,7 +295,7 @@ def load_and_exec_node(
                 flow_file=flow_file,
             )
 
-        flow_inputs = FlowExecutor._apply_default_value_for_input(flow.inputs, flow_inputs)
+        flow_inputs = FlowExecutor._process_input_values(flow.inputs, flow_inputs)
```
Comment (huaiyan) [inline src/promptflow/promptflow/executor/flow_executor.py:298]: Also need `_save_image_from_output` for single node run?

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 2]
```diff
@@ -776,6 +784,20 @@ def _apply_default_value_for_input(inputs: Dict[str, FlowInputDefinition], line_
...
 
+    @staticmethod
+    def _convert_image_to_bytes(inputs: Dict[str, FlowInputDefinition], line_inputs: Mapping) -> Dict[str, Any]:
+        updated_inputs = dict(line_inputs or {})
+        for key, value in inputs.items():
+            if value.type == ValueType.IMAGE:
+                updated_inputs[key] = Image.from_file(updated_inputs[key])
+        return updated_inputs
+
+    def _save_image_from_output(self, output: dict, output_dir: str):
```
Comment (huaiyan) [inline src/promptflow/promptflow/executor/flow_executor.py:795]: Add todo for writing each node's output file.
For now, if we have multi nodes, seems like subsequent nodes' input will be bytes in `RunTracker`, we may need change it to relative path later.
No need to bother for this PR.

Comment (huaiyan) [inline src/promptflow/promptflow/executor/flow_executor.py:795]: Seems like we didn't try to save output image file for each node.
Just to make sure that's one known issue.

Comment (lumoslnt) [reply src/promptflow/promptflow/executor/flow_executor.py:795]: We will save the intermediate images when serializing output.

**Review by thy09 — APPROVED** [review]

**Review by huaiyan — APPROVED** [review]

**Pull request merged**